src/integrations/admin/workouts-integration.php

```diff
@@ -258,10 +326,82 @@ private function get_update_premium_notice() {
 	 * @return bool Returns true when Premium is enabled and the version is below 17.7.
 	 */
 	private function should_update_premium() {
-		$premium_version = YoastSEO()->helpers->product->get_premium_version();
+		$premium_version = $this->product_helper->get_premium_version();
 		return $premium_version !== null && version_compare( $premium_version, '17.7-RC1', '<' );
 	}
 
+	/**
+	 * Check whether the Premium subscription has expired.
+	 *
+	 * @return bool Returns true when Premium subscription has expired.
+	 */
+	private function has_premium_subscription_expired() {
+		$subscription = $this->addon_manager->get_subscription( WPSEO_Addon_Manager::PREMIUM_SLUG );
+
+		return ( isset( $subscription->expiry_date ) && ( strtotime( $subscription->expiry_date ) - time() ) < 0 );
+	}
+
+	/**
+	 * Check whether the Premium subscription is activated.
+	 *
+	 * @return bool Returns true when Premium subscription is activated.
+	 */
+	private function has_premium_subscription_activated() {
+		return $this->addon_manager->has_valid_subscription( WPSEO_Addon_Manager::PREMIUM_SLUG );
+	}
+
+	/**
+	 * Returns the upsell/update copy to show in the card buttons.
+	 *
+	 * @return string Returns a string with the upsell/update copy for the card buttons.
+	 */
+	private function get_upsell_text() {
+		if ( $this->product_helper->is_premium() && $this->should_update_premium() ) {
+			if ( $this->has_premium_subscription_expired() ) {
+				return sprintf(
+					/* translators: %s: expands to 'Yoast SEO Premium'. */
+					__( 'Renew %s', 'wordpress-seo' ),
+					'Yoast SEO Premium'
+				);
+			}
+			if ( $this->has_premium_subscription_activated() ) {
+				return sprintf(
+					/* translators: %s: expands to 'Yoast SEO Premium'. */
+					__( 'Update %s', 'wordpress-seo' ),
+					'Yoast SEO Premium'
+				);
+			}
+			return sprintf(
+				/* translators: %s: expands to 'Yoast SEO Premium'. */
+				__( 'Activate %s', 'wordpress-seo' ),
+				'Yoast SEO Premium'
+			);
+		}
+
+		// Use the default defined in the component.
+		return '';
+	}
+
+	/**
+	 * Returns the upsell/update link to show in the card buttons.
+	 *
+	 * @return string Returns a string with the upsell/update link for the card buttons.
+	 */
+	private function get_upsell_link() {
+		if ( $this->product_helper->is_premium() && $this->should_update_premium() ) {
+			if ( $this->has_premium_subscription_expired() ) {
+				return 'https://yoa.st/workout-renew-notice';
+			}
+			if ( $this->has_premium_subscription_activated() ) {
+				return \wp_nonce_url( \self_admin_url( 'update.php?action=upgrade-plugin&plugin=wordpress-seo-premium/wp-seo-premium.php' ), 'upgrade-plugin_wordpress-seo-premium/wp-seo-premium.php' );
+			}
+			return 'https://yoa.st/workouts-activate-notice-myyoast';
+		}
+
+		// Use the default defined in the component.
+		return '';
+	}
+
```

Comment:
```suggestion
	private function get_upsell_link() {
		if ( ! $this->product_helper->is_premium() && $this->should_update_premium() ) {
	                return '';
                }
                
	        if ( $this->has_premium_subscription_expired() ) {
		        return 'https://yoa.st/workout-renew-notice';
	        }
	        
	        if ( $this->has_premium_subscription_activated() ) {
		        return \wp_nonce_url( \self_admin_url( 'update.php?action=upgrade-plugin&plugin=wordpress-seo-premium/wp-seo-premium.php' ), 'upgrade-plugin_wordpress-seo-premium/wp-seo-premium.php' );
	        }
	        
	        return 'https://yoa.st/workouts-activate-notice-myyoast';
        }
```

We could also consider making this into one method, together with `get_upsell_text` as we are doing the exact same checks here.